htmlreport/info_from_cmd.py

```diff
@@ -4,56 +4,49 @@
 import locale
 
 
-def get_results(line, path, file, blame_options):
+def git_blame(line, path, file, blame_options):
+    git_blame_dict = {
+        'author': 'Unknown',
+        'author_mail': '---',
+        'date': '---'
+    }
     head, tail = os.path.split(file)
     if head != "":
         path = head
+
     try:
         os.chdir(path)
     except:
-        return ""
+        return git_blame_dict
+
     try:
-        result = subprocess.check_output(f'git blame -L {str(line)}{" -w" if "-w" in blame_options else ""}{" -M" if "-M" in blame_options else ""} --porcelain -- {file}')
+        result = subprocess.check_output('git blame -L %d%s%s --porcelain -- %s' % (line, " -w" if "-w" in blame_options else "", " -M" if "-M" in blame_options else "", file))
         result = result.decode(locale.getpreferredencoding())
     except:
-        return ""
+        return git_blame_dict
 
-    return result
+    if result.startswith('fatal'):
+        return git_blame_dict
 
+    author = result[result.find('author') + 6:result.find('author-mail') - 1]
```

Comment:
This hard coding makes me scared. Some day they could change the output from git blame and then this will not work well at all. What is the input string? It would feel more robust if `author` and `author-mail` could be reordered and if other fields can be inserted without having to rewrite this code.